src/wallet/wallet.h

```diff
@@ -58,7 +58,7 @@ static const unsigned int MAX_FREE_TRANSACTION_CREATE_SIZE = 1000;
 //! Size of witness cache
 //  Should be large enough that we can expect not to reorg beyond our cache
 //  unless there is some exceptional network disruption.
-static const unsigned int WITNESS_CACHE_SIZE = COINBASE_MATURITY;
+static const unsigned int WITNESS_CACHE_SIZE = MAX_REORG_LENGTH + 1;
```

Comment:
If we decided to pick a larger `MAX_REORG_LENGTH`, this ensures that the witness cache will work regardless. The local witness cache would simply be extended until it reached the larger maximum. The larger maximum would not be a safe re-org limit however until after `MAX_REORG_LENGTH - COINBASE_MATURITY` blocks after the user has restarted post-upgrade.